src/app/fdctl/keygen.c

```diff
@@ -6,6 +6,23 @@
 #include <stdio.h>
 #include <sys/stat.h>
 
+void
+keygen_cmd_args( int *    pargc,
+                 char *** pargv,
+                 args_t * args) {
+  char * usage = "usage: keygen <identity|vote>";
+  if( FD_UNLIKELY( *pargc < 1 ) ) FD_LOG_ERR(( "%s", usage ));
+
+  if( FD_LIKELY( !strcmp( *pargv[ 0 ], "identity" ) ) ) args->keygen.key = "identity";
```

Comment:
Use some constants for this,

#define KEYGEN_KEY_IDENTITY              (0UL)
#define KEYGEN_KEY_VOTE_ACCOUNT (1UL)

And save to a ulong rather than this string.